app/models/service.rb

```diff
@@ -180,13 +173,14 @@ def shutdown_guest
     queue_group_action(:shutdown_guest, last_index, -1, delay_for_action(last_index, :stop))
   end
 
-  def power_states_match?(action)
-    all_states_match?(action) ? update_power_status(action) : false
+  def power_states_match?(action, states = nil)
+    all_states_match?(action, states) ? update_power_status(action) : false
   end
 
-  def all_states_match?(action)
-    return true if composite? && (power_states.uniq == map_power_states(action))
-    return true if atomic? && (power_states[0] == POWER_STATE_MAP[action])
+  def all_states_match?(action, states = nil)
+    states = power_states if states.nil?
```

Comment:
`states ||= power_states` or can `states == false`?

Reply:
`states ||= power_states` will work, states is always either a populated or empty array.